networkx/algorithms/flow/networksimplex.py

```diff
@@ -540,7 +546,7 @@ def update_potentials(i, p, q):
     # Infeasibility and unboundedness detection
     ###########################################################################
 
-    if any(x[i] != 0 for i in range(-n, 0)):
+    if any(abs(d) > tolerance for d in x):
```

Comment:
As discussed in a previous comment the `range(-n, 0)` was important here, since `x` may have length greater than `n`. So instead, do `for d in x[-n:]`.